src/libpriv/rpmostree-core.c

```diff
@@ -2586,6 +2447,7 @@ relabel_in_thread_impl (RpmOstreeContext *self,
                                        NULL, NULL, NULL);
 
   ostree_repo_commit_modifier_set_devino_cache (modifier, cache);
+  ostree_repo_commit_modifier_set_sepolicy (modifier, self->sepolicy);
```

Comment:
To answer your question, I can't think of a good reason right now *why* it wasn't done this way. Definitely https://github.com/ostreedev/ostree/issues/1165 would've been a blocker, though clearly it could've been fixed back then too. Anyway, this is definitely better!